exercises/pascals-triangle/test/test_pascals_triangle.c

```diff
@@ -76,25 +74,74 @@ static void test_four_rows(void)
       {1, 2, 1, 0},
       {1, 3, 3, 1}
    };
-   size_t **r = create_triangle(4);
-   TEST_ASSERT_TRUE(check(4, expected, r));
-   free_triangle(r, 4);
+   size_t **actual = create_triangle(4);
+   TEST_ASSERT_TRUE(check_triangle(4, expected, actual));
+   free_triangle(actual, 4);
+}
+
+static void test_five_rows(void)
+{
+   TEST_IGNORE();
+   size_t expected[5][5] = {
+      {1, 0, 0, 0, 0},
+      {1, 1, 0, 0, 0},
+      {1, 2, 1, 0, 0},
+      {1, 3, 3, 1, 0},
+      {1, 4, 6, 4, 1}
+   };
+   size_t **actual = create_triangle(5);
+   TEST_ASSERT_TRUE(check_triangle(5, expected, actual));
+   free_triangle(actual, 5);
+}
+
+static void test_six_rows(void)
+{
+   TEST_IGNORE();
+   size_t expected[6][6] = {
+      {1, 0, 0, 0, 0, 0},
+      {1, 1, 0, 0, 0, 0},
+      {1, 2, 1, 0, 0, 0},
+      {1, 3, 3, 1, 0, 0},
+      {1, 4, 6, 4, 1, 0},
+      {1, 5, 10, 10, 5, 1},
+   };
+   size_t **actual = create_triangle(6);
+   TEST_ASSERT_TRUE(check_triangle(6, expected, actual));
+   free_triangle(actual, 6);
 }
 
-static void test_negative_rows(void)
```

Comment:
Should we simplify the example now that this case doesn't exist? Better yet, should we update the interface so that the row count is unsigned?

Reply:
So I've done a bit of a cleanup all round:
- Changed the triangle values type from `size_t` to `uint8_t`, changed the rows similarly as suggested.
- Changed array loop iterators and `row` argument type from `int` to `size_t`
- Made some of the arguments and internal variable names a bit nicer.
- Added error checking on the `calloc()` use.... And have just realised that our `check` test helper function should test for `NULL` - will add now.

Reply:
Thanks. Great cleanup all around. There were a lot of little oddities here.